pkg/multimodelconfig/configmap.go

```diff
@@ -0,0 +1,110 @@
+package multimodelconfig
+
+import (
+	"encoding/json"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var logger = log.Log.WithName("MultiModelConfig")
+
+type ConfigsDelta struct {
+	updatedCfg map[string]string
+	deletedCfg map[string]string
+}
+
+type ModelConfig struct {
+	fileName    string
+	fileContent *ModelDefinition
+}
+
+type ModelDefinition struct {
+	StorageUri string `json:"storageUri"`
+	Framework  string `json:"framework"`
+}
+
+func NewConfigsDelta(updatedCfg[]ModelConfig, deletedCfg []ModelConfig) (*ConfigsDelta, error) {
+	var configsDelta ConfigsDelta
+	var err error
```

Comment:
Consider typing your return types to avoid these vars

```func NewConfigsDelta(updatedCfg[]ModelConfig, deletedCfg []ModelConfig) (delta *ConfigsDelta, err error)```